### PR TITLE
Check repository label for possible non-string value

### PR DESCRIPTION
**Pull request by poveu**
In my case line 151 was causing an error with integer given to strtolower function, so I added the missing casting to string

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19526) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @poveu. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by VladimirZaets** [comment]
Hi @poveu, please, sign CLA, otherwise, we can't process your pull request

**Comment by orlangur** [comment]
@poveu also, please provide exact steps to reproduce a problem on a vanilla Magento instance.

**Comment by poveu** [comment]
@magento-engcom-team give me {2.3-develop} instance

**Comment by orlangur** [comment]
@poveu such instance is not really helpful as there is no CLI access.

Please investigate on your local instance with something like

```
if (!is_string($label)) {
    var_dump($label);
    die;
}
```

**Comment by poveu** [comment]
> @poveu such instance is not really helpful as there is no CLI access.
> 
> Please investigate on your local instance with something like
> 
> ```
> if (!is_string($label)) {
>     var_dump($label);
>     die;
> }
> ```

Right! Sorry.
var_dump'ing $label gives only "int(0)".
However var_dump'ing $composerData['repositories'] gives:

> array(2) {
>   [0]=>
>   array(2) {
>     ["type"]=>
>     string(8) "composer"
>     ["url"]=>
>     string(25) "https://repo.magento.com/"
>   }
>   ["firegento_extendedimport2"]=>
>   array(2) {
>     ["type"]=>
>     string(3) "vcs"
>     ["url"]=>
>     string(54) "https://github.com/firegento/FireGento_ExtendedImport2"
>   }
> }

**Comment by orlangur** [comment]
@poveu https://github.com/magento/magento2-community-edition/blob/master/composer.json#L12 wondering how it wasn't found earlier O>o

File: dev/tools/UpgradeScripts/pre_composer_update_2.3.php
Comment (orlangur) [inline dev/tools/UpgradeScripts/pre_composer_update_2.3.php]: There is no sense to search for "mage" in numeric string, please change to

```
if (is_string($label) && strpos(strtolower($label), 'mage') !== false || strpos($repo['url'], '.mage') !== false) {
```

and do commit amend + force push.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
@poveu could you please squash changes into a single commit as I requested? Just tell if you need any assistance with that.

**Comment by poveu** [comment]
@orlangur actually I have no idea how to squash two commits if I created them via GitHub website GUI... ;)
I could create new PR from my fork if it is easier, but this will duplicate this PR.

**Review by orlangur — APPROVED** [review]
@poveu yeah, just noticed there is no real branch behind. Two commits is lesser evil than 2 PRs ;)

**Review by orlangur — APPROVED** [review]

**Review by orlangur — CHANGES_REQUESTED** [review]
@poveu, sorry, we cannot process contribution from an `unknown repository`.

Could you please re-create this PR using a proper fork? Just apply similar changes via GitHub UI and do no remove fork until PR is merged.

**Comment by poveu** [comment]
@orlangur haha sure, no problem :) #19711

**Pull request closed without merging**
